### PR TITLE
Change the default of `include_global_state` from true to false for snapshot restores

### LINKED ISSUE #18569: Change `include_global_state` default to `false`
It is trappy to restore an index and suddenly have your global state change, eg see https://github.com/elastic/elasticsearch/issues/18559

I think state should only be restored if specifically requested.  We should change the `include_global_state` default to `false`

### PR DESCRIPTION
**Pull request by dakrone**
Resolves #18569

### CODE REVIEW

File: docs/reference/migration/migrate_5_0/settings.asciidoc [hunk 1]
```diff
@@ -277,3 +277,8 @@ should be used instead.
 
 The setting `bootstrap.mlockall` has been renamed to
 `bootstrap.memory_lock`.
+
+==== Snapshot settings
+
+The default setting `include_global_state` for taking and restoring snapshots
+has been changed from `true` to `false`.
```
Comment (ywelsch) [inline docs/reference/migration/migrate_5_0/settings.asciidoc]: That's not true? `CreateSnapshotRequest` still has `includeGlobalState = true`.

Comment (ywelsch) [inline docs/reference/migration/migrate_5_0/settings.asciidoc]: The snapshot / restore documentation also needs to be updated if we want to make this change.

**Comment by ywelsch** [comment]
Left some comments. I think the intention of #18569 was to change the `includeGlobalState` default only for restores.

File: docs/reference/migration/migrate_5_0/settings.asciidoc (hunk 1 above)
Comment (dakrone) [inline docs/reference/migration/migrate_5_0/settings.asciidoc]: Yes, sorry! I was spacing yesterday and this change is only intended to change the default for restores, I'll update this.

**Comment by dakrone** [comment]
@ywelsch you're right, I just made the migration note incorrect. I've updated the title of this as well as the migration note to be clear about that, thanks for looking!

**Comment by ywelsch** [comment]
@dakrone can you also update the snapshot / restore documentation (`snapshots.asciidoc`)? The first section there reads `By default, all indices in the snapshot as well as cluster state are restored.` and continues talking about `include_global_state`.

**Comment by dakrone** [comment]
@ywelsch good idea, I've updated it.

File: docs/reference/modules/snapshots.asciidoc [hunk 2]
```diff
@@ -323,33 +323,40 @@ POST /_snapshot/my_backup/snapshot_1/_restore
 -----------------------------------
 // CONSOLE
 
-By default, all indices in the snapshot as well as cluster state are restored. It's possible to select indices that
-should be restored as well as prevent global cluster state from being restored by using `indices` and
-`include_global_state` options in the restore request body. The list of indices supports
-<<search-multi-index-type,multi index syntax>>. The `rename_pattern` and `rename_replacement` options can be also used to
-rename index on restore using regular expression that supports referencing the original text as explained
+By default, all indices in the snapshot are restored, the cluster state is not
```
Comment (ywelsch) [inline docs/reference/modules/snapshots.asciidoc]: maybe add a BUT somewhere.

File: docs/reference/modules/snapshots.asciidoc [hunk 3]
```diff
@@ -323,33 +323,40 @@ POST /_snapshot/my_backup/snapshot_1/_restore
...
+restored. It's possible to select indices that should be restored as well as
+prevent global cluster state from being restored by using `indices` and
```
Comment (ywelsch) [inline docs/reference/modules/snapshots.asciidoc]: this should read: It's possible .... as well as **allow** global cluster state to be restored ...

**Comment by ywelsch** [comment]
2 more minor comments, LGTM o.w.. Feel free to push once addressed.

**Pull request merged**